sysinfo.sh

```diff
@@ -0,0 +1,130 @@
+#!/bin/bash
+
+###########################################################
+#
+# Author: ctrlbyte
+# Sending_Grounds Bash Scripting Challenge
+#
+# To stay in the spirit of ProLUG and Killercoda labs, 
+# script serves as a psuedo-ansible "gather-facts" clone, 
+# collecting system info from specified hosts. 
+# Assumption is that "control node" running already has 
+# setup passwordless SSH login to target nodes.
+# 
+# Requirements:
+# - Catches signals
+# - Has a help/usage statement
+# - Uses named Exit codes
+# - Has arg parsing
+# - Posts it to this repo here
+# - and the script of course works
+# 
+###########################################################
+
+# Named exit codes
+EXIT_SUCCESS=0
+EXIT_INVALID_ARGS=1
+EXIT_SIGNAL_RECEIVED=2
+
+# Print help/usage
+usage() {
+    echo "Usage: $0 [options] <hostname1> <hostname2> ..."
+    echo "  Collects system info from specified hosts"
+    echo "Options:"
+    echo "  -q          Run quick scan (default)"
+    echo "  -f          Run full scan"
+    echo "  -h          Show help/usage"
```

Comment:
Only thing I'd suggest is maybe using a **heredoc** here, as that'll give you more options and make it cleaner, something akin to:

```bash
cat<<EOF 
Usage: $0 [options] <hostname1> <hostname2>
Type: Non-mutating (information only)

Collects system info from specified hosts

Options:
    -q                        Run quick scan (default)
    -f                         Run full scan
    -h                        Show help/usage
EOF
```

That'll allow you less copy/paste with regards to the echo..., but this also is totally legit and works.